common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2922,6 +2920,10 @@ def _copy_subdag(self, user_id, destination_version, block_key, source_blocks, d
                 raw=True,
                 block_defaults=new_block.defaults
             )
+            # _new_block doesn't copy the original_usage data needed for library blocks:
+            if new_block.edit_info.original_usage:
+                destination_block.edit_info.original_usage = new_block.edit_info.original_usage
+                destination_block.edit_info.original_usage_version = new_block.edit_info.original_usage_version
```

Comment:
Are there other places where we could end up needing to preserve this info when calling `_new_block`? (And if so, wouldn't it be better to allow to pass additional `edit_info` to `_new_block`?)

For example, duplicating a library content block within a unit in studio doesn't seem to preserve the `original_usage` - I haven't worked much with the modulestore code and I don't know what function from split this ends up using, but but it ends up missing in the event in my tests:

```json
{
  "username": "staff",
   "event_type": "edx.librarycontentblock.content.assigned",
   "ip": "178.19.210.162",
   "agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:36.0) Gecko/20100101 Firefox/36.0",
   "host": "sandbox4.opencraft.com",
   "referer": "http://sandbox4.opencraft.com/courses/course-v1:TestX+TST-CL-1+now/courseware/9d92d7c4495c4025acebe89069d0950a/59d291d746214f17b2359712bbf67c99/1",
   "accept_language": "en-US;q=1.0, en;q=0.5",
   "event": {
    "previous_count": 0,
     "max_count": 1,
     "location": "block-v1:TestX+TST-CL-1+now+type@library_content+block@0fcc4e0e18144755a2edb392f939105d",
     "added": [
      {
        "original_usage_version": null,
         "descendants": [
          
        ],
         "usage_key": "block-v1:TestX+TST-CL-1+now+type@html+block@474244681cd84e72ba1b7c22efd52a58",
         "original_usage_key": null
      }
    ],
     "result": [
      {
        "original_usage_version": null,
         "descendants": [
          
        ],
         "usage_key": "block-v1:TestX+TST-CL-1+now+type@html+block@474244681cd84e72ba1b7c22efd52a58",
         "original_usage_key": null
      }
    ]
  },
   "event_source": "server",
   "context": {
    "course_user_tags": {
      
    },
     "user_id": 4,
     "org_id": "TestX",
     "course_id": "course-v1:TestX+TST-CL-1+now",
     "path": "/courses/course-v1:TestX+TST-CL-1+now/courseware/9d92d7c4495c4025acebe89069d0950a/59d291d746214f17b2359712bbf67c99/"
  },
   "time": "2015-03-19T13:13:29.017528+00:00",
   "page": "x_module"
}
```